src/System.Management.Automation/engine/MshCommandRuntime.cs

```diff
@@ -3701,5 +3689,45 @@ internal void RemoveVariableListsInPipe()
                 _state.PSVariable.Remove(this.PipelineVariable);
             }
         }
+
+        /// <summary>
+        /// Stream variables are setup as ArrayLists, but we want this to be
+        /// transparent so that the following are indistinguishable:
+        /// 
+        ///     $ov = Write-Output @(1,2); $ov.GetType().FullName
+        ///
+        ///     Write-Output -OutVariable ov @(1,2); $ov.GetType().FullName
+        ///
+        /// So when we clean up internally, set the variables back to the
+        /// ordinary PowerShell return types
+        /// </summary>
+        /// <param name="variableName">the name of the variable to set</param>
+        /// <param name="varList">the variable value stored in the ArrayList</param>
+        private void SetNonArrayListVariable(string variableName, IList varList)
+        {
+            if (variableName == null)
+            {
+                return;
+            }
+
+            // No values -> return null
```

Comment:
Seems `return` is not correct term in comments here and below.

Reply:
Good point!